packages/mcfunction/src/parser/macro.ts

```diff
@@ -76,8 +74,15 @@ export function macro(): core.Parser<MacroNode> {
 			}
 		} while (nextChunk.length > 0)
 
-		// No actual macro
-		if (children.length < 3) {
+		// A line with no macros is invalid
+        let hasMacro = false
+        for (const child of children) {
+            if (child.type === 'mcfunction:macro_child/macro') {
+                hasMacro = true
+                break
```

Comment:
This can probably be compacted to `children.some(c => c.type === 'mcfunction:macro_child/macro')`